packages/core/src/App/Containers/Redirect/redirect.jsx

```diff
@@ -20,9 +21,16 @@ const Redirect = ({
     const url_params = new URLSearchParams(url_query_string);
     let redirected_to_route = false;
     const { is_dashboard } = React.useContext(PlatformContext);
+    const uses_allowed_language = Object.keys(getAllowedLanguages()).includes(currentLanguage);
+    const default_to_english = getURL(currentLanguage).replace(`&lang=${currentLanguage}`, '');
+    const new_signup = url_params.get('action') === 'signup';
 
     setVerificationCode(url_params.get('code'), url_params.get('action'));
 
+    if (new_signup && !uses_allowed_language) {
+        window.location.assign(default_to_english);
```

Comment:
We have `changeLanguage` function in `language.js`. Can we use that instead?

Reply:
Initially, I tried using the changeLanguage function in language.js. Although, it is a bit different than the one I am currently using. The one in language.js replaces the URL, and for some reason it doesn't let the modal pop-up anymore when that happens. I think this is because .replace simply replaces part of the link and refreshes the page (not 100% sure how it works). The modal wont pop-up anymore when you refresh the page, it needs to see the link kind of as a new instance. This is why I used the assign function here which does make a new modal pop-up.